server/proxy/pf_channels.c

```diff
@@ -74,7 +96,10 @@ void pf_OnChannelDisconnectedEventHandler(void* context,
 	}
 	else if (strcmp(e->name, RDPGFX_DVC_CHANNEL_NAME) == 0)
 	{
-		gdi_graphics_pipeline_uninit(((rdpContext*)pc)->gdi,
-		                             (RdpgfxClientContext*) e->pInterface);
+		gdi_graphics_pipeline_uninit(((rdpContext*)context)->gdi, (RdpgfxClientContext*) e->pInterface);
+	}
+	else if (strcmp(e->name, DISP_DVC_CHANNEL_NAME) == 0)
+	{
+		pc->disp = NULL;
```

Comment:
should we free stuff here as well ?

Reply:
No, disp context is freed in server side in any connection

Reply:
where?

Reply:
https://github.com/FreeRDP/FreeRDP/blob/91f3c277ccdd73af8c4876fc349992b18f47e08e/server/proxy/pf_server.c#L331